sfge/src/main/java/com/salesforce/graph/symbols/apex/SoqlQueryInfo.java

```diff
@@ -70,6 +74,7 @@ protected SoqlQueryInfo _merge(SoqlQueryInfo other) {
                 // query
                 this.securityEnforced
                         && other.securityEnforced, // Both portions need to be security enforced
+                this.userMode && other.userMode, // Both portions need to be user mode
```

Comment:
This isn't causing any test failures, but I'm not positive this is correct. Should it be an `||` instead of `&&`?
Is the `merge` method intended for merging two wholly separate queries, or merging an inner query with an outer query?

Reply:
`merge` is intended to merge information from two entirely different objects. This is for situations where multiple Read calls are made within a single path on the same object. Pretty sure there are other cases as well and I'm trying to remember.
Might be worth adding tests to see how they operate when a path has two queries on the same object and one of them has `WITH USER_MODE` and another doesn't.

Reply:
Have a look at `ObjectFieldUtilTest.java` and `ObjectFieldInfo.java`.